articles/why-you-should-use-celery-with-rabbitmq/index.md

```diff
@@ -0,0 +1,463 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/why-you-should-use-celery-with-rabbitmq/
+title: Why You Should Use Celery With RabbitMQ
+description: This article will provide an overview of task queues & message brokers, why a user should choose celery with rabbitmq, and finally an example that shows how to use the two technologies.
+author: edidiong-etuk
+date: 2021-04-04T00:00:00-14:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/why-you-should-use-celery-with-rabbitmq/hero.png
+    alt: Celery and RabbitMQ hero image
+---
+In today's technology space, there is an increase in event-driven architectures as companies develop solutions that require asynchronous communication between their microservices. Ingesting these events faster in system architecture and processing them enables system architectures to be persistent, resilient and allows batch processing of data. Celery and RabbitMQ are some tools used in these architectures. Choosing the right combination of tools and viewing an example of these tools that goes beyond the ["hello world"](https://docs.celeryproject.org/en/stable/getting-started/brokers/rabbitmq.html) is what this article cover.
+
+### Outline
+
+1. [What We Will Cover In This Article](#what-we-will-cover-in-this-article)
+2. [Understanding Celery & RabbitMQ](#understanding-celery-&-rabbitmq)
+    - [What is a Task Queue](#what-is-a-queue)
+    - [Examples of Task Queues](#examples-of-task-queues)
+    - [What is a Message Broker](#what-is-a-message-broker)
+    - [Examples of Message Brokers](#examples-of-message-brokers)
+3. [Why We Should Choose The Celery & RabbitMQ Combo](#why-we-should-choose-the-celery-&-rabbitmq-combo)
+4. [Setting up Celery with RabbitMQ](#setting-up-celery-with-rabbitmq)
+    - [Installation & Configuration](#installation-&-configuration)
+    - [Simple Use Case](#simple-use-case)
+5. [Conclusion](#conclusion)
+
+### What We Will Cover In This Article
+
+In this article, we’ll walk through what tasks queue are, see the examples of task queues (which includes celery), learn about message brokers. We will also look at why we need to use Celery and RabbitMQ while also seeing a basic tutorial that shows its installation and usage.
+
+### Understanding Celery & RabbitMQ
+
+Before going into why we should use Celery & RabbitMQ, let us understand what they are and what similar tools are available. Celery is classified as a *Task Queue* and RabbitMQ is classified as a *Message Broker*. Let's dive deeper into what these are.
+
+#### What is a Task Queue?
+
+A task queue is a data structure maintained by a job scheduler containing jobs to run.
+Task queue software also manages background work that must be executed outside of the usual HTTP request-response cycle. They are designed for asynchronous work.
```

Comment:
Please define what asynchronous work is, since a beginner might not be aware of async requests